### PR TITLE
enable exact balance listener option via metadata

### PR DESCRIPTION
**Pull request by bwallacef5**
**Please provide a description of this PR:**

Fixes https://github.com/istio/istio/issues/42374

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by bwallacef5** [comment]
currently only inbound is functional, but I am looking for input on the overall approach.

File: pilot/pkg/model/context.go [hunk 1]
```diff
@@ -645,14 +645,6 @@ type NodeMetadata struct {
 	// ExitOnZeroActiveConnections terminates Envoy if there are no active connections if set.
 	ExitOnZeroActiveConnections StringBool `json:"EXIT_ON_ZERO_ACTIVE_CONNECTIONS,omitempty"`
 
-	// InboundListenerExactBalance sets connection balance config to use exact_balance for virtualInbound,
-	// as long as QUIC, since it uses UDP, isn't also used.
-	InboundListenerExactBalance StringBool `json:"INBOUND_LISTENER_EXACT_BALANCE,omitempty"`
```
Comment (howardjohn) [inline pilot/pkg/model/context.go:650]: why remove this?

**Review by howardjohn — CHANGES_REQUESTED** [review]
https://github.com/istio/istio/issues/42374 is a user error. They should set ISTIO_META_ prefix. The code is correct

**Pull request closed without merging**